src/pages/home/sidebar/ChatSwitcherView.js

```diff
@@ -461,12 +466,47 @@ class ChatSwitcherView extends React.Component {
             }
         }
 
-        this.setState({
-            options: Array.from(matches),
-        });
+        let options = Array.from(matches);
+        if (options.length === 0) {
+            const searchStr = value;
+            if (searchStr && searchStr.includes('@') && Str.isValidEmail(searchStr)) {
+                options = [{
+                    text: searchStr,
+                    alternateText: searchStr,
+                    singleUserDM: true,
+                    type: CONST.OPTION_TYPE.PERSONAL_DETAIL,
+                    keyForList: searchStr,
+                    login: searchStr,
+                }];
+            }
+
+            const isPhoneNumber = /^[+|\d]+$/.test(searchStr);
+            if (searchStr && isPhoneNumber && Str.isValidPhone(searchStr)) {
+                const phoneNumber = searchStr.includes('+') ? searchStr : `+${this.props.countryCodeByIP}${searchStr}`;
+                options = [{
+                    text: phoneNumber,
+                    alternateText: phoneNumber,
+                    singleUserDM: true,
+                    type: CONST.OPTION_TYPE.PERSONAL_DETAIL,
+                    keyForList: phoneNumber,
+                    login: searchStr,
+                }];
+            }
+        }
+
+        this.setState({options});
     }
 
     render() {
+        let feedbackHeader = '';
+        let feedbackMessage = '';
+        if (this.state.usersToStartGroupReportWith.length === MAX_GROUP_DM_LENGTH) {
+            feedbackHeader = 'Maximum participants reached';
+            feedbackMessage = 'You\'ve reached the maximum number of participants for a group chat.';
+        } else if (this.state.search && this.state.options.length === 0) {
+            feedbackMessage = 'Don\'t see who you are looking for? Type their valid email/phone number to invite them.';
+        }
```

Comment:
This works well but I wonder how you feel about creating a new UI component for the feedback message and then inlining everything? I think it looks a bit cleaner and encourages us to reuse the feedback message style and also clearly define the states that make the messages appear.

```js
const maxParticipantsReached = this.state.usersToStartGroupReportWith.length === MAX_GROUP_DM_LENGTH;
const hasSelectableOptions =  this.state.search && this.state.options.length > 0;
```

And then broke this out into 3 separate blocks using a shared UI component called `FeedbackMessage`

```js
{!hasSelectableOptions && (
    <FeedbackMessage
        message={'blah...'}
    />
)}
{maxParticipantsReached && (
    <FeedbackMessage
        message={'blah...'}
        title="blah"
    />
)}
{hasSelectableOptions && !maxParticipantsReached && (
    ....
)}
```

Reply:
I'm neutral about creating another component just because I don't know where it can be reused at the moment and hence I'm unsure if it's preoptimization. Let me know if you feel otherwise and i can still create another component for FeedbackMessage. Although i'm unsure why `ChatSwitcherView` would call it 3 times rather than just once with the message it needs. I say that to prevent bugs, eg: say maxParticipantsReached is true and hasSelectableOptions is false. Then we could end up showing two messages which is not what we want. So i think the logic of creating the feedbackMessage would still remain and then we'd show the `FeedbackMessage` only if feedbackMessage has a value.

Reply:
Sure, I think if we used this pattern maybe one more time then we would want to create a component. Rule of three? And we have two already.

>Although i'm unsure why ChatSwitcherView would call it 3 times rather than just once with the message it needs

I just think it's easier to make sense out of what is happening. Set up your two conditions that you are looking for. Show what happens to the message for each condition with a clear name for each. Then no one has to parse the if/elseif or the extra variables + which one has a title and which one does... well above the context of where they are used. The alternative is nice and clean and tells a story.

It's fine if we don't want to do this now, but I think what you have is hard to read so that's the suggestion I had.

Reply:
> Sure, I think if we used this pattern maybe one more time then we would want to create a component. Rule of three?

Sounds good with me. Curious to pick your brain on this to understand how you think. Here is my usual thought pattern and you can let me know if there is something incorrect. I usually think of breaking something in a component if it can be reused in one other place. I don't really think of breaking it into another component if it's just used once in a single component (even if it can be called many times in one component).
Because,
```
{condition1 && (
    <FeedbackMessage
        message={'blah...'}
    />
)}
{condition2 && (
    <FeedbackMessage
        message={'blah...'}
        title="blah"
    />
)}
{condition3 && (
    <FeedbackMessage
        message={'blah...'}
        title="blah"
    />
)}
{!condition1 && !condition2 && !condition3 && (
    ....
)}
```
Still means that FeedbackMessage if shown is just shown once and then in that case it feels like the code would be easier to read like so,
```
let feedBackMessage = '';
if (condition1) {
    feedBackMessage = 'message1';
} else if (condition1) {
    feedBackMessage = 'message2';
} else if (condition3) {
    feedBackMessage = 'message3';
}

{feedBackMessage && (
    <FeedbackMessage
        message={feedBackMessage}
        title="blah"
    />
)}
{!feedBackMessage && (
    ....
)}
``` 

> Then no one has to parse the if/elseif or the extra variables + which one has a title and which one does

Is it generally a recommended pattern to keep lesser logic like this and instead have more variables like hasSelectableOptions and maxParticipantsReached and then have the render handle them accordingly rather than thinking about the least number of component calls? I was just curious if there is some general pattern i should improve on wrt react.

Reply:
I don't mind having extra variables set up if it helps describe a state later on. It's not really a React thing to me just a general readability thing and can help with refactoring or separating concerns in a complicated component like this.

I am also in the process of refactoring this anyway so I wont block on this. One thing that will happen eventually is the logic for these two "feedback messages" will not exist in the same place anymore. This will eventually just be a prop of "feedbackMessage" and "feedbackTitle". Breaking the if/else into two separate concerns will make it easier to see when the two messages show. One is only relevant to group chats the other is relevant to both group chats and new chats.

Reply:
Also, I think this kind of small helper text style will end up being used a lot. That's the only reason I think a component would be good here. But we cant wait on that.